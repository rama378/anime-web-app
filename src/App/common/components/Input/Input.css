.Input {
  width: 100%;
  font-weight: bold;
  font-family: var(--font-family);
  border: none;
  border-radius: 16px;
}

.Input::placeholder {
  font-weight: normal;
  font-style: italic;
  color: #606060;
}

.Input:focus {
  outline: none;
}

.Input__wrapper {
  position: relative;
}

.Input__wrapper svg {
  position: absolute;
  right: -8px;
  width: 30px;
  height: 30px;
  padding: 12px 0;
}

.Input--default {
  padding: 16px;
  font-size: var(--font-size-large);
}

.Input--large {
  padding: 16px;
  font-size: var(--font-size-larger);
}
