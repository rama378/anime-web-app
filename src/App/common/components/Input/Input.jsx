import PropTypes from 'prop-types';
import './Input.css';

const Input = (props) => {
	const { size, className, icon, ...rest } = props;

	return (
		<div className={`Input__wrapper ${className}`}>
			<input {...rest} className={`Input Input--${size}`} />
			{icon}
		</div>
	);
};

Input.propTypes = {
	size: PropTypes.string,
	className: PropTypes.string,
	icon: PropTypes.any,
};

Input.defaultProps = {
	size: 'default',
	className: null,
	icon: null,
};

export default Input;
