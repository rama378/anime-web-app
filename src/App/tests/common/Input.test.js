import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Input from '../../common/components/Input/Input';
import {ReactComponent as SearchIcon} from "../../common/assets/images/icon/search.svg";

const renderer = new ShallowRenderer();

describe('<Input />', () => {
  it('should render and match the snapshot no prop given', () => {
    renderer.render(<Input />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot given prop size large', () => {
    renderer.render(<Input size="large" />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot given prop className Search__input', () => {
    renderer.render(<Input className="Search__input" />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot given prop className Search__input', () => {
    renderer.render(<Input icon={<SearchIcon />} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
