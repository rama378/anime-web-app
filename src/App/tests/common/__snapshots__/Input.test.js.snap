// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render and match the snapshot given prop className Search__input 1`] = `
<div
  className="Input__wrapper Search__input"
>
  <input
    className="Input Input--default"
  />
</div>
`;

exports[`<Input /> should render and match the snapshot given prop className Search__input 2`] = `
<div
  className="Input__wrapper null"
>
  <input
    className="Input Input--default"
  />
  <ForwardRef(SvgSearch) />
</div>
`;

exports[`<Input /> should render and match the snapshot given prop size large 1`] = `
<div
  className="Input__wrapper null"
>
  <input
    className="Input Input--large"
  />
</div>
`;

exports[`<Input /> should render and match the snapshot no prop given 1`] = `
<div
  className="Input__wrapper null"
>
  <input
    className="Input Input--default"
  />
</div>
`;
